### ISSUE TITLE
onBlur, onChange, onDrop, onDragStart not passed into Fields component

### ISSUE THREAD

**Issue by huan086** [erikras/redux-form#2535]
See https://github.com/erikras/redux-form/pull/2161#pullrequestreview-14746339

#2161 introduced handler caching. While API for `Field` has been fixed by passing these handlers to `input`, the `Fields` has not been fixed

**Comment by hpurmann** [comment]
Any news on this? We're getting `input.onDragStart` is marked as required in `Component`, but its value is `undefined`.

**Comment by hpurmann** [comment]
@erikras @ghostd Can you elaborate on why these props are marked as required? Do we really have to pass empty functions?

**Comment by ghostd** [comment]
Hi,

It's a bug. [Documentation](https://github.com/erikras/redux-form/blob/e63bd4934f08238dedfeecdb8ab812d3f4bf546a/docs/api/Field.md#ondragstart--event--void-optional) mentions it as optional

So we should align the [prop types](https://github.com/erikras/redux-form/blob/b2cef7191ced63fbe16841fdf84625622cac5d17/src/propTypes.js#L60) (and the [flow version](https://github.com/erikras/redux-form/blob/639d45edf57f99d2d55d49d0b063c00dcf5198bb/src/FieldProps.types.js.flow#L34)?)

**Comment by hpurmann** [comment]
Ah ok. So then someone would need to remove `isRequired` from `onDragStart` and `onDrop`, right? Is `onBlur`, `onChange`, `onFocus` correct?
```javascript
  onDragStart: func,
  onDrop: func,
```
I'm not familiar with this codebase though. Would this be correct?

**Comment by ghostd** [comment]
I think it would match the documentation

**Comment by gustavohenke** [comment]
Are folks still having issues with the original problem reported here?

**Comment by BeLi4L** [comment]
@gustavohenke yes, unfortunately the issue is still here, Fields doesn't pass `onDragStart` in `input`.

I reproduced the bug in this jsfiddle: https://jsfiddle.net/osjc5g2f/

**Comment by jakebellacera** [comment]
Still getting this issue. Can we update these prop types? Not sure why we state that they're required when they're clearly not.
